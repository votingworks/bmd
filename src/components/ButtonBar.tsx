import styled from 'styled-components'

interface Props {
  secondary?: boolean
}

const ButtonBar = styled('nav')<Props>`
  order: ${({ secondary }) => (secondary ? '-1' : undefined)};
  display: flex;
  padding: 0.5rem;
  background: #455a64;
  border-bottom: 1px solid darkgrey;
  justify-content: space-between;
  align-items: center;
  & > :first-child {
    order: 2;
  }
  & > * {
    margin: 0.25rem
    flex: 1;
  }
  @media print {
    display: none;
  }
`

export default ButtonBar
