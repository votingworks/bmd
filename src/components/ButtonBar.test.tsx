import React from 'react'
import { render } from '@testing-library/react'

import ButtonBar from './ButtonBar'

it(`renders ButtonBar`, () => {
  const { container } = render(<ButtonBar>foo</ButtonBar>)
  expect(container.firstChild).toMatchSnapshot()
})

it(`renders ButtonBar with natural order and primary button separated`, () => {
  const { container } = render(
    <ButtonBar separatePrimaryButton naturalOrder>
      <div>one</div>
      <div>two</div>
      <div>three</div>
    </ButtonBar>
  )
  expect(container.firstChild).toMatchSnapshot()
})
