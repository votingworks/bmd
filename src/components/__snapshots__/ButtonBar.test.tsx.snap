// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ButtonBar 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.5rem;
  background: #455a64;
  border-bottom: 1px solid darkgrey;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > :first-child {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

.c0 > * {
  margin: 0.25rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media print {
  .c0 {
    display: none;
  }
}

<nav
  class="c0"
>
  foo
</nav>
`;
